python/whylogs/api/whylabs/session/session_manager.py

```diff
@@ -0,0 +1,285 @@
+import logging
+import time
+import uuid
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from enum import Enum
+from typing import Dict, List, Optional, Union, cast
+
+import requests as r
```

Comment:
requests is not an existing whylogs core dependency, we will need to add this to pyproject.toml, can do with `poetry add requests` and then need to update the poetry lock file.